independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/DirectoryClassPathElement.java

```diff
@@ -72,7 +72,15 @@ public String getPath() {
                 @Override
                 public URL getUrl() {
                     try {
-                        return file.toUri().toURL();
+                        URI uri = file.toUri();
+                        // the URLClassLoader doesn't add trailing slashes to directories, so we make sure we return
+                        // the same URL as it would to avoid having QuarkusClassLoader return different URLs
+                        // (one with a trailing slash and one without) for same resource
+                        if (uri.getPath().endsWith(File.separator)) {
```

Comment:
Does Windows generate the URI with backslashes? Even though \ is the file seperator / is the URI seperator, so I can't remember which one it produces?

Reply:
I was hoping CI would tell, but it failed for a different reason. Just looking around on the internet it seems that for Windows it continues to use `/`, so I'll update this and see what happens

Reply:
Looks like CI passed.